wrapper/src/test/java/software/amazon/jdbc/HikariPooledConnectionProviderTest.java

```diff
@@ -141,7 +141,9 @@ void testConnectWithDefaultMapping() throws SQLException {
     Properties props = new Properties();
     props.setProperty(PropertyDefinition.USER.name, user1);
     props.setProperty(PropertyDefinition.PASSWORD.name, password);
-    try (Connection conn = provider.connect(protocol, mockDialect, mockTargetDriverDialect, mockHostSpec, props)) {
+    ConnectionInfo connectionInfo =
+        provider.connect(protocol, mockDialect, mockTargetDriverDialect, mockHostSpec, props);
+    try (Connection conn = connectionInfo.getConnection()) {
```

Comment:
Curious, does it make sense to make `ConnectionInfo` implement `AutoCloseable`?
```java
try (ConnectionInfo connectionInfo = provider.connect(protocol, mockDialect, mockTargetDriverDialect, mockHostSpec, props);
Connection conn = connectionInfo.getConnection()) {
  // do stuff
}
If I understood correctly the connection is being opened in this line 
```java
ConnectionInfo connectionInfo =
        provider.connect(protocol, mockDialect, mockTargetDriverDialect, mockHostSpec, props);
```
so this line could throw exceptions, and should be wrapped in try-catch.

Reply:
I reworked the test code and use try-finally to control connections.